rclrs/src/executor.rs

```diff
@@ -1,45 +1,60 @@
-use crate::{rcl_bindings::rcl_context_is_valid, Node, RclReturnCode, RclrsError, WaitSet};
+use crate::{
+    rcl_bindings::rcl_context_is_valid, Context, ContextHandle, IntoNodeOptions, Node, RclrsError,
+    WaitSet,
+};
 use std::{
     sync::{Arc, Mutex, Weak},
     time::Duration,
 };
 
 /// Single-threaded executor implementation.
-pub struct SingleThreadedExecutor {
+pub struct Executor {
+    context: Arc<ContextHandle>,
     nodes_mtx: Mutex<Vec<Weak<Node>>>,
```

Comment:
With the removal of `remove_node()`, it should now be the responsibility of the `Executor` to clean up old `Node`s. So to avoid leaking the `Node` allocations, we should filter those out, possibly within `spin` or `spin_once`.

Reply:
Good point, added: [d047214](https://github.com/ros2-rust/ros2_rust/pull/428/commits/d0472147e83d91a14dd62c0a9ea6479409b9b2f0)

Reply:
Thanks, looks good!